File: Backend/src/main/java/upp/demo/model/Transaction.java
package upp.demo.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import upp.demo.enumeration.TransactionStatus;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Table
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private LocalDateTime timestamp;

    @ManyToOne
    private User initiator;

    @ManyToOne
    private Book book;

    @Column
    private TransactionStatus status;

    @Column(unique=true)
    private String successCode;

    @Column(unique=true)
    private String failedCode;

    @Column(unique=true)
    private String errorCode;
}

File: Frontend/literarySocietyFront/src/app/model/transaction.ts
export class Transaction {
    id:number;
    timestamp:string;
    initiatorId:number;
    bookId:number;
    status:number;

    constructor(id:number, timestamp:string, initiatorId:number, bookId:number, status:number) {
        this.id = id;
        this.timestamp = timestamp;
        this.initiatorId = initiatorId;
        this.bookId = bookId;
        this.status = status;
    }
}
  